src/lib/storage/proposal/devicegraph_generator.rb

```diff
@@ -92,6 +93,19 @@ def provide_space(volumes, initial_graph, disk_analyzer)
           result_graph
         end
 
+        def refine_volumes(volumes, space_maker)
+          refined = volumes.deep_dup
+          deleted_swap = space_maker.deleted_partitions.detect do |part|
+            part.id == ::Storage::ID_SWAP
+          end
+          swap_volume = volumes.detect { |vol| vol.mount_point == "swap" }
+          if deleted_swap && swap_volume
+            swap_volume.uuid = deleted_swap.uuid
+            swap_volume.label = deleted_swap.label
+          end
+          refined
+        end
```

Comment:
So far we only generate one swap volume. But it's true this part should be agnostic from that fact. I will modify it so it tries to reuse as many swaps UUIDs as possible.